executor/read/control_instruction.rs

```diff
@@ -4,16 +4,40 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
  */
 
+use compiler::VariablePosition;
+
 use crate::{
     batch::{FixedBatch, FixedBatchRowIterator},
     read::{
-        collecting_stage_executor::CollectedStageIterator,
-        pattern_executor::{BranchIndex, ExecutorIndex},
+        collecting_stage_executor::{CollectedStageIterator, CollectorEnum},
         stream_modifier::StreamModifierResultMapper,
+        BranchIndex, ExecutorIndex,
     },
     row::MaybeOwnedRow,
 };
 
+#[derive(Debug)]
+pub(super) enum ControlInstruction {
+    // Control instructions
+    PatternStart(PatternStart),
+    RestoreSuspension(RestoreSuspension),
+    ReshapeForReturn(ReshapeForReturn),
+    Yield(Yield),
+
+    ExecuteImmediate(ExecuteImmediate),
+
+    MapBatchToRowsForNested(MapBatchToRowsForNested),
+    ExecuteNegation(ExecuteNegation),
+
+    ExecuteBranch(ExecuteBranch),
```

Comment:
Renamed `ExecuteDisjunction` because this is only one branch

Reply:
can we call it `ExecuteDisjunctionBranch`? To keep it in line with Negation being right there (and dont have to recall that branch is now standard terminology in disjunctions. Also branching has other meanings, like if/else with conditions, which we dont have)